chia/types/blockchain_format/program.py

```diff
@@ -279,7 +279,7 @@ def run_as_generator(
 
         err, ret = run_generator(
             self._buf,
-            serialized_args,
+            bytes(serialized_args),
```

Comment:
Is it strictly necessary to do this cast for `run_generator`? Maybe we can add `Union[bytes, bytearray]` to this argument and everything still works without the cast.

Reply:
That said, even without that optimization, this is probably still faster.

Reply:
yes, this is strictly necessary. PyO3 `PyBytes` does not support the buffer protocol. It would need a similar treatment as `parse_rust()` in order to accept a `memoryview` or `bytearrat`.

`PyBytes` fail without this conversion